docs/parse_argparse.py

```diff
@@ -0,0 +1,118 @@
+import glob
+import subprocess
+import os
+import textwrap 
+import shutil
+
+path_scripts_2 = '../docs/source/scripts_2'
+if os.path.exists(path_scripts_2):
+    shutil.rmtree(path_scripts_2) 
+
+for dir_name in glob.glob("../autometa/**/*.py", recursive=True): 
+    if "__" not in dir_name: #to not include __init__ files
+        print(dir_name)
+        with open (dir_name) as fh_read:
+            path_temp_write = '../docs/temp_write.py'
+            fh_write = open(path_temp_write, 'w+')            
+            for line in fh_read:
+                if line.strip() == 'import argparse':
+                    line = line.strip()
+                    fh_write.write(line + '\n')
+                    break
+            # Reads text until the end of the block:
+            for line in fh_read:  # This keeps reading the file
+                if line.strip() == 'args = parser.parse_args()':
+                    line = line.strip()
+                    fh_write.write(line + '\n')
+                    break
+                line = line.strip()
+                fh_write.write(line + '\n')
+            fh_write.close()
```

Comment:
Are you trying to write all the lines between `import argparse` and `args = parser.parse_args()` (inclusive)? The code above does that I believe, but it is a bit hard to read. You could have a separate variable that controls whether lines are written. Something like this:

```python
fh_write = open(path_temp_write, 'w+')
recording = False
for line in fh_read:
    if line.rstrip() == 'import argparse':
        recording = True
    
    if recording:
        fh_write.write(line)

    if line.rstrip() == 'args = parser.parse_args()':
        recording = False
fh_write.close()
```

Note, in the above the checks for the start and end of the recording section are also before and after the `fh_write.write` statement. This is so that the last line is written to the output file. Also, I do have two `line.rstrip()` statements, but the number of lines is less, so I like it this way.

Reply:
Thanks @jason-c-kwan this was really helpful !